src/core/router/info.h

```diff
@@ -522,6 +523,11 @@ class RouterInfo : public RouterInfoTraits, public RoutingDestination
   ///   (and subsequently sign the RI with)
   void CreateBuffer(const PrivateKeys& private_keys);
 
+  /// @brief Verify RI signature
+  /// @return Success of signature verification
+  /// @throws std::length_error if unsigned buffer length is below minimum
+  bool Verify();
```

Comment:
This should also be implemented outside of the test.

Reply:
Are you referring to throwing the length exception? If so, I can add that into the buffer refactor. If not, what do you mean?

Reply:
Actually use this function `Verify()` outside of the test.

Reply:
Oh, understood. Maybe at the end of contruction (after buffer refactor)? Will look for other places that make sense too.

Reply:
Oh, understood. Maybe at the end of construction (after buffer refactor)? Will look for other places that make sense too.